src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionCodeFixProvider.PragmaBatchFixHelpers.cs

```diff
@@ -123,71 +126,59 @@ private static class PragmaBatchFixHelpers
                 bool includeEndTokenChange,
                 CancellationToken cancellationToken)
             {
-                var newDocument = await pragmaAction.GetChangedDocumentAsync(includeStartTokenChange, includeEndTokenChange, cancellationToken).ConfigureAwait(false);
-                
-                // Update the diagnostics spans based on the text changes.
-                var textChanges = await newDocument.GetTextChangesAsync(currentDocument, cancellationToken).ConfigureAwait(false);
-                foreach (var textChange in textChanges)
-                {
-                    UpdateDiagnosticSpans(diagnostics, currentDiagnosticSpans, textChange);
-                }
-
-                return textChanges;
+                var newDocument = await pragmaAction.GetChangedDocumentAsync(includeStartTokenChange, includeEndTokenChange, cancellationToken).ConfigureAwait(false);                
+                return await newDocument.GetTextChangesAsync(currentDocument, cancellationToken).ConfigureAwait(false);
             }
 
-            private static async Task UpdateDiagnosticSpansAsync(Document currentDocument, Document newDocument, ImmutableArray<Diagnostic> diagnostics, Dictionary<Diagnostic, TextSpan> currentDiagnosticSpans, CancellationToken cancellationToken)
+            private static void UpdateDiagnosticSpans(ImmutableArray<Diagnostic> diagnostics, Dictionary<Diagnostic, TextSpan> currentDiagnosticSpans, IEnumerable<TextChange> textChanges)
             {
-                // Update the diagnostics spans based on the text changes.
-                var textChanges = await newDocument.GetTextChangesAsync(currentDocument, cancellationToken).ConfigureAwait(false);
-                foreach (var textChange in textChanges)
-                {
-                    UpdateDiagnosticSpans(diagnostics, currentDiagnosticSpans, textChange);
-                }
-            }
-
-            private static void UpdateDiagnosticSpans(ImmutableArray<Diagnostic> diagnostics, Dictionary<Diagnostic, TextSpan> currentDiagnosticSpans, TextChange textChange)
-            {
-                var isAdd = textChange.Span.Length == 0;
-                Func<TextSpan, bool> isPriorSpan = span => span.End <= textChange.Span.Start;
-                Func<TextSpan, bool> isFollowingSpan = span => span.Start >= textChange.Span.End;
-                Func<TextSpan, bool> isEnclosingSpan = span => span.Contains(textChange.Span);
+                Func<TextSpan, TextChange, bool> isPriorSpan = (span, textChange) => span.End <= textChange.Span.Start;
```

Comment:
TextSpan, TextSpan? it doesnt actually use any information except span from TextChange.